modules/ingest-common/src/main/java/org/elasticsearch/ingest/common/JsonProcessor.java

```diff
@@ -75,7 +87,8 @@ public JsonProcessor create(Map<String, Processor.Factory> registry, String proc
                                     Map<String, Object> config) throws Exception {
             String field = ConfigurationUtils.readStringProperty(TYPE, processorTag, config, "field");
             String targetField = ConfigurationUtils.readStringProperty(TYPE, processorTag, config, "target_field", field);
-            return new JsonProcessor(processorTag, field, targetField);
+            boolean addToRoot = ConfigurationUtils.readBooleanProperty(TYPE, processorTag, config, "add_to_root", false);
```

Comment:
if `addToRoot` is true then maybe fail if target field is specified? Not a big a fan of silently ignoring this.